crates/ruff_benchmark/benches/red_knot.rs

```diff
@@ -27,13 +27,17 @@ static EXPECTED_DIAGNOSTICS: &[&str] = &[
     // We don't support `*` imports yet:
     "error[unresolved-import] /src/tomllib/_parser.py:7:29 Module `collections.abc` has no member `Iterable`",
     // We don't support terminal statements in control flow yet:
+    "error[annotation-with-invalid-expression] /src/tomllib/_parser.py:57:71 Invalid expression in type expression",
```

Comment:
There are several type annotations on this line, and it's hard to tell which annotation this is referring to without counting the columns. Could we improve the error message so it specifies which variable or annotation the diagnostic is complaining about?

Reply:
Including arbitrary expressions is something that I'm always worried about because it can lead to very long messages, but maybe David can come up with something that avoids this. 

What should help here is when we have rich diagnostics that also show a code frame

Reply:
When I wrote that error message, I was looking for something like a `ast::Expr::human_readable_name()` method, but didn't find anything.

I'm happy to improve this error message though once we agree that this is even the right approach to begin with.

> What should help here is when we have rich diagnostics that also show a code frame

:+1:

Reply:
> What should help here is when we have rich diagnostics that also show a code frame

That should help for sure, but it's also useful to have important information in the summary line for when a user is using `--output-format=concise`. I found `--output-format=concise` to be very useful when filing PRs to upgrade projects to Ruff 0.8, because there were many new violations on some codebases, and it was very hard to scroll through them all when using the default verbose output.